### DISCUSSION TITLE
ScreenToWorldCoordinate

### DISCUSSION THREAD

**Discussion by simonsland** [google/filament#5665] [Q&A]
Recently, I've tried to transform screen coordinate to world coordinate. However, I found that the result world coordinate isn't equal to the expect coordinate, but is very closed. Below is the code, is there any mistake I make to get the the correct coordinate?
`  math::vec3<float> origin;
    math::vec3<float> dir;
    
    _manipulator->getRay(x, y, &origin, &dir);
    
    math::vec3<float> result;
    
    result = origin + dir * abs(z - self.cameraController.eyePosition.z);`

**Comment by bejado** [comment]
I'm not sure what `z` or `self.cameraController.eyePosition.z` are, but note we do something similar [here](https://github.com/google/filament/blob/main/libs/camutils/src/Manipulator.cpp#L278):

```
*result = mEye + dir * t;
```
